File: Leetcode/python/sol203.py
#leetcode.com/problems/remove-linked-list-elements/

# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        prev = ListNode(-1, head)
        rslt = prev
        current = head
        while current:
            if current.val == val:
                prev.next = current.next
            else:
                prev = prev.next
            current = current.next
        return rslt.next

File: Leetcode/cpp/sol203.cpp

//https://leetcode.com/problems/remove-linked-list-elements/description/
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {

        ListNode dummy(-1, head);
        ListNode* prev = &dummy;
        ListNode* current = head;

        while(current){
            if(current->val == val){
                prev->next = current->next;
            }else{
                prev = prev->next;
            }
            current = current->next;
        }

        return dummy.next;
    }
};